text/0143-versioning-and-stability.md

```diff
@@ -0,0 +1,128 @@
+# OpenTelemetry versioning and stability
+
+OpenTelemetry is a large project with strict compatibility requirements. This proposal defines the stability gurantees offered by OpenTelemetry, along with a versioning and lifecycle proposal which defines how we meet those requirements.
+
+Language implementations are expected to follow this proposal exactly, unless a langague or package manager convention interferes significantly. Implementations must take this cross-language proposal, and produce a language-specific proposal which details how these requirements will be met.
+
+## Design goals
+
+**Ensure that end users stay up to date with the latest release.**  
+We want all users to  stay up to date with the latest version of OpenTelemetry. We do not want to create hard breaks in support, of any kind, which leave users stranded on older versions. It must always be possible to upgrade to the latest minor version of OpenTelemetry, without creating compilation or runtime errors.
+
+**Never create a dependency conflict between packages which rely on different versions of OpenTelemetry. Avoid breaking all stable public APIs.**  
+Backwards compatibility is a strict requirement. Instrumentation APIs cannot create a version conflict, ever. Otherwise, OpenTelemetry cannot be embedded in widely shared libraries, such as web frameworks. Theoretically, APIs can be deprecated and eventually removed, but this is a process measured in years.
+
+**Allow for multiple levels of package stability within the same release.**  
+Provide maintainers a clear process for developing new, experimental APIs alongside stable APIs. DIfferent packages within the same release will have different levels of stability.
+
+## Relevant architecure
+![drawing](img/0143_cross_cutting.png)
+
+At the highest architectural level, OpenTelemetry is organized into signals. Each signal provides a specialized form of observability. For example, tracing, metrics, and baggage are three seperate signals. Signals share a common subsystem – context propagation – but they function independently from each other.
+
+Each signal provides a mechanism for software to describe itself. A codebase, such as an API handler or a database client, takes a dependency on various signals in order to describe itself. OpenTelemetry instrumentation code is then mixed into the other code within that codebase. This makes OpenTelemetry a **cross-cutting concern** - a piece of software which must be mixed into many other pieces of software in order to provide value. Cross-cutting concerns, by their very nature, violate a core design principlie – seperation of concerns. As a result, OpenTelemetry requires extra care and attention to avoid creating issues for the codebases which depend upon it for instrumentaion.
```

Comment:
> the codebases which depend upon it for instrumentaion

Nit: something a la `depend upon it for self-description` may be better. Applications don't use Otel for instrumentation. As you said at the beginning of this section `takes a dependency on various signals in order to **describe itself**`

Reply:
Not sure what you mean by "Applications don't use Otel for instrumentation". I would say they do that a lot, when the application is adding manual instrumentation to its own code using otel APIs. do you not call that instrumentation?

Reply:
I agree with John, how is it that applications don't instrument themselves? 

Anyways, I did try to reword this to clarify that it is the cross-cutting nature of the APIs which make them so heavily depended upon.

"OpenTelemetry requires extra care and attention to avoid creating issues for the codebases which depend upon these cross-cutting APIs."

Let me know if that is better.

Reply:
It's nitpicking the verbiage, feel free to ignore :) 

First sentence is "Each signal provides a mechanism for software **to describe itself**".
Last sentence is "codebases which depend upon it for **instrumentaion**"

But they both talk about one and the same action/process/whatever.